staking-contract/contracts/StakingRewardSystem.sol

```diff
@@ -0,0 +1,221 @@
+            _addRewardToken(_rewardTokens[i]);
+        }
+    }
+
+    // Pause the contract
+    function pause() external onlyOwner {
+        _pause();
+    }
+
+    // Unpause the contract
+    function unpause() external onlyOwner {
+        _unpause();
+    }
+
+    // Function to stake tokens
+    function stake(uint256 _amount, address _rewardToken) external nonReentrant whenNotPaused {
+        if (_amount <= 0) revert InvalidAmount();
+        if (_rewardToken == address(0)) revert InvalidInput("Reward token address cannot be zero");
+        if (!isRewardToken[_rewardToken]) revert InvalidRewardToken();
+        stakingToken.safeTransferFrom(msg.sender, address(this), _amount);
+        Stake storage userStake = userStakes[msg.sender][_rewardToken];
+        userStake.amount += _amount;
+        userStake.startTime = block.timestamp;
+        emit Staked(msg.sender, _amount, _rewardToken);
+    }
+
+    // Function to withdraw staked tokens
+    function withdraw(uint256 _amount, address _rewardToken) external nonReentrant whenNotPaused {
+        if (_amount <= 0) revert InvalidAmount();
+        if (_rewardToken == address(0)) revert InvalidInput("Reward token address cannot be zero");
+        if (!isRewardToken[_rewardToken]) revert InvalidRewardToken();
+        Stake storage userStake = userStakes[msg.sender][_rewardToken];
+        if (userStake.amount < _amount) revert WithdrawAmountExceedsStake();
+
+        uint256 penalty = 0;
+        if (block.timestamp < userStake.startTime + minStakingPeriod) {
+            penalty = (_amount * earlyWithdrawalPenalty) / 1e18;
+            if(penalty+_amount>userStake.amount) revert InvalidInput("Withdrawl amount is greater");
+            _amount -= penalty;
+            accumulatedPenalties += penalty;
+        }
+        userStake.amount -= _amount;
+        stakingToken.transfer(msg.sender, _amount);
+        emit Withdrawn(msg.sender, _amount, penalty, _rewardToken);
+    }
+
+  
+    // Function to claim rewards based on staked tokens
+    function claimReward(address _rewardToken) external nonReentrant whenNotPaused {
+        if (_rewardToken == address(0)) revert InvalidInput("Reward token address cannot be zero");
+        if (!isRewardToken[_rewardToken]) revert InvalidRewardToken();
+        Stake storage userStake = userStakes[msg.sender][_rewardToken];
+        if (userStake.amount == 0) revert NoStakeFound();
+        if (block.timestamp <= userStake.startTime + minStakingPeriod) revert CannotClaimRewardYet();
+
+        uint256 rewardAmount = _calculateReward(userStake);
+        IERC20(_rewardToken).safeTransfer(msg.sender, rewardAmount);
+        userStake.startTime = block.timestamp;
+        emit RewardClaimed(msg.sender, rewardAmount, _rewardToken);
+    }
+
+    // Internal function to calculate the reward based on staking duration
+    function _calculateReward(Stake storage userStake) internal view returns (uint256) {
```

Comment:
Parameteric values should be underscored,, make sure to follow the conventions once communicated

Reply:
Why passing storage variable?
Use calldata for function parameters.